src/ng/compile.js

```diff
@@ -2555,10 +2574,10 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
             break;
 
           case '=':
-            if (optional && !attrs[attrName]) {
-              return;
-            }
             parentGet = $parse(attrs[attrName]);
+
+            if (parentGet === noop && optional) break;
```

Comment:
this is wrong as in the example `watch` is a function and `$parse(fn) === fn` for al functions `fn`